src/umpire/strategy/mixins/Inspector.cpp

```diff
@@ -47,13 +47,13 @@ Inspector::registerAllocation(
   ResourceManager::getInstance().registerAllocation(ptr, record);
 }
 
-void 
+util::AllocationRecord* 
 Inspector::deregisterAllocation(void* ptr)
 {
   auto record = ResourceManager::getInstance().deregisterAllocation(ptr);
 
   m_current_size -= record->m_size;
-  delete record;
+  return record;
```

Comment:
Won't this cause a leak?

Reply:
No way around it really, we have to break the abstraction of the Inspector to allow this information to leak out 😞

Reply:
Okay, this is all the more reason to place all of this into a single helper function that checks m_strategy, produces UMPIRE_ERROR, and does a delete

Reply:
See comment above.